pkg/controller/apiserver/apiserver_controller_test.go

```diff
@@ -686,5 +688,123 @@ var _ = Describe("apiserver controller tests", func() {
 			Expect(instance.Status.Conditions[2].Message).To(Equal("Not Applicable"))
 			Expect(instance.Status.Conditions[2].ObservedGeneration).To(Equal(generation))
 		})
+		Context("Management cluster reconciliation", func() {
+			BeforeEach(func() {
+				// Create the ManagementCluster CR needed to configure
+				// a management cluster for a multi-cluster setup
+				managementCluster := &operatorv1.ManagementCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "tigera-secure",
+					},
+					Spec: operatorv1.ManagementClusterSpec{
+						TLS: &operatorv1.TLS{
+							SecretName: render.VoltronTunnelSecretName,
+						},
+					},
+				}
+				Expect(cli.Create(ctx, managementCluster)).NotTo(HaveOccurred())
+
+				// Create the APIServer CR needed to jumpstart the reconciliation
+				// for the api server
+				err := cli.Create(ctx, &operatorv1.APIServer{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "tigera-secure",
+						Namespace: common.OperatorNamespace(),
+					},
+				})
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Should reconcile multi-cluster setup for a management cluster for a single tenant", func() {
+				// Create the tunnel secret in operator namespace
+				// This will be created by Manager controller in tigera-operator namespace
```

Comment:
"In a production cluster, this would be created... " instead of "This will be created"